core/tests/performance_framework/perf_services.py

```diff
@@ -58,21 +60,64 @@ class SeleniumPerformanceDataFetcher(object):
     # XHR requests made post initial page load will not be recorded.
     WAIT_DURATION_SECS = 3
 
-    def __init__(self, browser=DEFAULT_BROWSER_SOURCE):
+    BASE_URL = 'http://localhost:%d' % feconf.PERFORMANCE_TESTS_SERVER_PORT
+
+    LIBRARY_URL_SUFFIX = '/library'
+    EDITOR_URL_SLICE = '/create/'
+    PLAYER_URL_SLICE = '/explore/'
+    PROFILE_URL_SLICE = '/profile/'
+    LOGIN_URL_SUFFIX = '/_ah/login'
+    ADMIN_URL_SUFFIX = '/admin'
+    DASHBOARD_URL_SUFFIX = '/dashboard'
+    SPLASH_URL_SUFFIX = '/splash'
+    SIGNUP_URL_SUFFIX = '/signup'
+    DUMMY_URL = '/robots.txt'
+
+    def __init__(self, browser=DEFAULT_BROWSER_SOURCE, preload_options=None):
+        """preload_options:
+            login: True if login is required.
+            enable_explorations: True if explorations needs to be enabled.
+            enable_collections: True if collections needs to be enabled.
+            create_exploration: True if exploration needs to be created.
+        """
         if browser in self.SUPPORTED_BROWSER_SOURCES:
             self.browser = browser
         else:
             error_msg = 'Unsupported browser specified: %s' % browser
             raise ValueError(error_msg)
 
+        self.preload_options = preload_options
+        self.exploration_url = None
+        self.dev_appserver_login_cookie = None
+
+        random_id = random.randint(1, 100000)
+        self.email = 'test%d@example.com' % random_id
+        self.username = 'username%d' % random_id
+        self.username = self.preload_options.get('username', self.username)
+
+        driver = self._setup_driver()
+        if preload_options['login']:
+            self._setup_login(driver)
+        elif preload_options['enable_explorations']:
+            self._setup_enable_explorations(driver)
+        elif preload_options['enable_collections']:
+            self._setup_enable_collections(driver)
+        elif preload_options['create_exploration']:
+            self._setup_create_exploration(driver)
+        self._stop_driver(driver)
+
     def load_url(self, page_url):
         """Loads the specified url, resulting in server-side caching of related
         resources. Used to obtain a uniform and warm-cache state for the tests,
         similar to a typical production server state.
         """
         driver = self._setup_driver(proxy=None, use_proxy=False)
 
-        driver.get(page_url)
+        get_page_url = page_url
+        if self.exploration_url:
```

Comment:
There was a design issue that came up, usually we provide the page_url from the test class, but since the exploration doesn't exist yet we can't. Hence, I needed a do it separately. I have refactored this approach now.